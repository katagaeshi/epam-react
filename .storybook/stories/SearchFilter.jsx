import React from 'react';
import {
  Provider,
  connect
} from 'react-redux';
import { storiesOf } from '@storybook/react';

import configureStore from '../../public/store';

import SearchFilter from '../../public/components/ui/SearchFilter';

import { setSearchBy } from '../../public/actions';

import storeInitialState from '../../public/store/initialState.json';

const { store } = configureStore(storeInitialState);


const mapStateToProps = state => {
  return {
    options: ['TITLE', 'GENRE'],
    checked: state.searchPanel.filter,
  };
};

const mapDispatchToProps = dispatch => ({
  onUpdate(changeEvent) {
    dispatch(setSearchBy(changeEvent.target.value));
  },
});

const SearchFilterContainer = connect(mapStateToProps, mapDispatchToProps)(SearchFilter);

storiesOf('SearchFilter', module)
  .addDecorator(story => <Provider store={store}>{story()}</Provider>)
  .add('with 2 options', () => (
      <SearchFilterContainer
        idPrefix='RBG_2_options'
        checked='TITLE'
      />
    )
  );
