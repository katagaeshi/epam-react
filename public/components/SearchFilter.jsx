import React from 'react';
import PropTypes from 'prop-types';
import RadioButtonGroup from './RadioButtonGroup';

const labelText = 'SEARCH BY';

const SearchFilter = props => (
  <div>
    <label htmlFor="SearchFilter">{labelText}
      <RadioButtonGroup id="SearchFilter" {...props} />
    </label>
  </div>
);

SearchFilter.propTypes = {
  options: PropTypes.arrayOf(PropTypes.string),
  checked: PropTypes.string,
};

SearchFilter.defaultProps = {
  options: ['Title', 'Director'],
  checked: 'Title',
};

export default SearchFilter;
