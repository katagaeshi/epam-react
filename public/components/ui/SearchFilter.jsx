// @flow

import React from 'react';
import injectSheet from 'react-jss';
import RadioButtonGroup from './RadioButtonGroup';

const styles = {
  searchFilter: {
    'grid-row': '3',
    'grid-column': '1 / 1',
  },
};

const filterText = 'SEARCH BY';

type Props = {
  options: Array<string>,
  idPrefix: string,
  classes: {
    searchFilter: string,
  },
};

const SearchFilter = (props: Props) => (
  <div className={props.classes.searchFilter}>
    <span>{filterText}</span>
    <RadioButtonGroup idPrefix="SearchFilter" {...props} />
  </div>
);

SearchFilter.defaultProps = {
  options: ['TITLE', 'GENRE'],
  idPrefix: 'SearchFilter-',
};

export default injectSheet(styles)(SearchFilter);
