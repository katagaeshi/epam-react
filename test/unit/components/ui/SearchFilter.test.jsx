import SearchFilter from 'public/components/ui/SearchFilter';

describe('SearchFilter', () => {
  it('Snapshot', () => {
    const tree = shallow(<SearchFilter
      onUpdate={() => {}}
    />);
    expect(tree).toMatchSnapshot();
  });
});
