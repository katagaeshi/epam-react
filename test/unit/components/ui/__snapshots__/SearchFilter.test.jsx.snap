// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFilter Snapshot 1`] = `
<SearchFilter
  classes={
    Object {
      "searchFilter": "SearchFilter-searchFilter-0-1-1",
    }
  }
  idPrefix="SearchFilter-"
  onUpdate={[Function]}
  options={
    Array [
      "TITLE",
      "GENRE",
    ]
  }
/>
`;
